iphone/cli/hooks/frameworks.js

```diff
@@ -552,13 +559,13 @@ class FrameworkIntegrator {
 	 * @access private
 	 */
 	addLinkFrameworkBuildPhase(frameworkInfo, fileRefUuid) {
-		let frameworkPackageName = frameworkInfo.name + '.framework';
-		var buildFileUuid = this._builder.generateXcodeUuid();
+		const frameworkPackageName = frameworkInfo.packageName;
+		const buildFileUuid = this._builder.generateXcodeUuid();
 		this._xobjs.PBXBuildFile[buildFileUuid] = {
 			isa: 'PBXBuildFile',
 			fileRef: fileRefUuid,
 			fileRef_comment: frameworkPackageName,
-			platformFilter: 'ios'
+			// platformFilter: 'ios' // FIXME: Is this needed?
```

Comment:
'platformFilter'  ensures that if included framework/xcframework is for 'ios' only or 'macos' only or iOS+macOS both (default).

So if a .framework supports ios only (which is general case), above line should be there.
In case of .xcframework, we need to check if it contains mac library or not. If not, above line should be there.